File: src/main/java/site/hirecruit/hr/domain/anonymous/entity/AnonymousEntity.java
package site.hirecruit.hr.domain.anonymous.entity;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

/**
 * @version 1.0.0
 * @author 전지환
 */
@Table
@Entity(name = "anonymous")
@Getter @Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class AnonymousEntity {

    @Id
    @Column(name = "anonymous_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long anonymousId;

    @NotNull
    @Column(name = "anonymous_uuid")
    private String anonymousUUID;

    @NotNull
    @Column(name = "name")
    private String name;

    @NotNull
    @Column(name = "email")
    private String email;

    @NotNull
    @Column(name = "email_certified")
    private boolean emailCertified;

}

File: src/main/java/site/hirecruit/hr/domain/anonymous/entity/AnonymousEntity.kt
package site.hirecruit.hr.domain.anonymous.entity

import javax.persistence.*

@Entity(name = "anonymous")
class AnonymousEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "anonymous_id")
    val anonymousId: Long?,

    @Column(name = "anonymous_uuid")
    private val anonymousUUID: String,

    @Column(name = "name")
    private val name: String,

    @Column(name = "email")
    private val email: String,

    @Column(name = "email_certified")
    private val emailCertified: Boolean? = false
) {
}